receiver/pluginreceiver/receiver.go

```diff
@@ -34,6 +34,8 @@ type Receiver struct {
 	logger         *zap.Logger
 	createService  createServiceFunc
 	service        Service
+
+	doneChan chan struct{}
```

Comment:
Does a wait group make more sense here? Maybe it makes it so you don't have to init a new channel in tests?

I think either one works so just wanted to bring it up, not pushing for it.

Reply:
Wait group could work too, for sure. I personally prefer the channel here just because it means I don't have to do the whole extra goroutine pattern in shutdown to utilize the context.